### PR TITLE
Issue #4475 fix

### PR DESCRIPTION
**Pull request by nixgc**
This is a change requested for list.js and block_events.js in #4475

## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Issue Resolved: [#4475](https://github.com/google/blockly/issues/4475)
### Proposed Changes  
This change is what was asked to be done with the deprecated code for block events and block provides. The block events that needed to be updated to what was asked is done and the block provides were removed as asked.

#### Behavior Before Change
Blockly works normally in all playgrounds.

#### Behavior After Change
Nothing changed after updating the Block Events and removing the Block Provides. Everything in the playgrounds continues to work fine.

### Reason for Changes
This change was done because some of the code being used here is old and needed to be updated or removed. I know that it was not a major need for change so I decided to take it on. The change follows what was asked to be done.

### Test Coverage
These changes need to be made because it allows Blockly's code that came from years ago to up-to-date and keeps it from being unused. 

### Documentation
I tested the code by running each of the playgrounds on Desktop Firefox and Desktop Safari. Since these did not make major changes I simply tested the blocks by creating simple functions and code to test and make sure that everything still runs properly. 

### Additional Information
There is no additional information needed.

### CODE REVIEW

**Review by moniika — COMMENTED** [review]
Looks good, just a few minor comments.

File: blocks/lists.js [hunk 1]
```diff
@@ -856,6 +855,7 @@ Blockly.Blocks['lists_split'] = {
    * @param {!Element} xmlElement XML storage element.
    * @this {Blockly.Block}
    */
+
```
Comment (moniika) [inline blocks/lists.js:858]: Please remove extra newline here

File: core/events/block_events.js [hunk 2]
```diff
@@ -568,6 +568,7 @@ Blockly.registry.register(Blockly.registry.Type.EVENT, Blockly.Events.CREATE,
 Blockly.registry.register(Blockly.registry.Type.EVENT, Blockly.Events.DELETE,
     Blockly.Events.Delete);
 Blockly.registry.register(Blockly.registry.Type.EVENT, Blockly.Events.CHANGE,
-    Blockly.Events.Change);
+    Blockly.Events.BlockChange);
 Blockly.registry.register(Blockly.registry.Type.EVENT, Blockly.Events.MOVE,
     Blockly.Events.Move);
+
```
Comment (moniika) [inline core/events/block_events.js:574]: Are there 2 newlines here? If so, then there should just be one.

**Review by moniika — COMMENTED** [review]
Some minor comments about newline. Otherwise, all that's left is for you to address the CLA agreement from the google-cla bot

**Comment by nixgc** [comment]
@googlebot I fixed it.

**Comment by nixgc** [comment]
Hello, so I've completed all of the minor fixes that needed to be done to clean up the code. I am confused about why it is that I am having an issue with the CLA agreement. I had signed the CLA agreement a while back and I confirmed this earlier. I worked on this issue by myself so I am confused about what it is telling me that I have co-authors. Is there another way around this?

**Comment by moniika** [comment]
> Hello, so I've completed all of the minor fixes that needed to be done to clean up the code. I am confused about why it is that I am having an issue with the CLA agreement. I had signed the CLA agreement a while back and I confirmed this earlier. I worked on this issue by myself so I am confused about what it is telling me that I have co-authors. Is there another way around this?

My first guess is that some of your git commits had a different email account than the one that you signed the CLA agreement for. When I look at the author of the first commit, that one looks different than all the subsequent commits.
https://github.com/google/blockly/pull/4758/commits/45d391ade696ee60ef71157adc624ea58b7428e6

**Comment by nixgc** [comment]
@googlebot I fixed it.

**Comment by moniika** [comment]
@nixgc I'm sorry the google-cla bot is giving so much trouble.
Could you try just squashing your commits and force pushing them to your branch? I think that way it will have the email account that the google-cla bot recognizes on the commit.

To squash you can use the command `git rebase -i`. Here's a [site](https://www.git-tower.com/learn/git/faq/git-squash/) with instructions if you're not famliar with that git command.

**Comment by moniika** [comment]
@nixgc When I look into more info I see that your first commit has the email "graysonnix@Graysons-MacBook-Pro.local" which doesn't have the CLA signed for it. Your options are either:
a) rebase your commits with the email address that signed the CLA ("61979456+nixgc@users.noreply.github.com")
b) add this other email address to your CLA

**Comment by moniika** [comment]
@nixgc Rather than fighting the bot, I'm going to go ahead and merge since you've acknowledged that it you did sign the agreement and it's pretty clear that all the commits are from you.

**Pull request merged**